### PR TITLE
Negative output

### PR DESCRIPTION
**Pull request by BWStearns**
The average function was pretty seriously broken. The atan2 function would return negative responses for the lower half of the unit circle that were `(desired_answer - TAU)`. I'll publish a new version of the crate since the crate is actually broken as published.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -9,33 +12,37 @@ pub use crate::chrono::avg_time_of_day;
...
         x_pos_sum = x_pos_sum + c;
         y_pos_sum = y_pos_sum + s;
     }
 
     let avg_x_pos = x_pos_sum / F::from(len).unwrap();
     let avg_y_pos = y_pos_sum / F::from(len).unwrap();
-
     // Get the angle of the average position
-    let avg_angle = avg_y_pos.atan2(avg_x_pos);
+    let atan2 = avg_y_pos.atan2(avg_x_pos);
```
Comment (Spindel) [inline src/lib.rs:38]: The compiler will probably fold this anyhow, but I'd rephrase it as:
```rust
let mut avg_angle = avg_y_pos.atan2(avg_x_pos);
// atan2 range is -pi to +pi, thus we adjust negative values with tau 
if avg_angle < F::zero() {
   avg_angle += tau
}
```

Comment (BWStearns) [reply src/lib.rs:38]: Good call.
